### PR TITLE
most recent dropxstrategy

### PR DESCRIPTION
**Pull request by ryu-bu**
dx14.lfr not wortking

### CODE REVIEW

File: lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py [hunk 1]
```diff
@@ -22,183 +23,175 @@ def reduce_mapping_options(self) -> None:
...
-                        ):
-
-                            if nonmeter_in_pred1 and nonmeter_in_pred2:
-                                # this is continuous
-                                pass  # temp
-                            else:
-                                # not continuous
-                                pass
+                                    print("not continuous")
+                                    pass
```
Comment (rkrishnasanka) [inline lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py:169]: @ryu-bu can you remove the commented-out code and put in a cleanly formatted comment saying what needs to happen ? 

You're printing out a statement saying not continuous but I'm not sure why that would be the case

File: lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py [hunk 2]
```diff
@@ -22,183 +23,175 @@ def reduce_mapping_options(self) -> None:
...
             # Rule 6 – if plus is sown between two nodes that has % in pred, then its droplet merging
             # Rule 7 – TBD Rule for droplet splitting
         # Finally just reduce the total number of mapping options if greater than 1
         super().reduce_mapping_options()
 
-    # recursive function do go through all the predecessors
     def __search_predecessors(self, fignode_id, search_type):
-        # check if the node has predecessors
-        does_exist = False
+        fignode = self._fig.get_fignode(fignode_id)
```
Comment (rkrishnasanka) [inline lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py:177]: Add a method description, install: https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

File: lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py [hunk 3]
```diff
@@ -22,183 +23,175 @@ def reduce_mapping_options(self) -> None:
         # input_fignodes = self._fig.get_input_fignodes
         for fignode_id in self._fig.nodes:
             fignode = self._fig.get_fignode(fignode_id)
+
+            # check if construction node
+            if ConstructionNode(fignode.id).is_explictly_mapped:
+                print("this is map")
```
Comment (rkrishnasanka) [inline lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py:29]: What is this supposed to mean ?

**Review by rkrishnasanka — CHANGES_REQUESTED** [review]
@ryu-bu overall it seems fine, I want you to split up the logic for each of the rules into different methods so that it becomes more manageable in the future

File: lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py (hunk 3 above)
Comment (ryu-bu) [reply lfr/netlistgenerator/v2/gen_strategies/dropxstrategy.py:29]: that's for if it's explicitly mapped or not. I'll comment out the print statement

**Review by rkrishnasanka — APPROVED** [review]
This is good, in general, you should use the return type and argument annotations for all the methods you create, or else no one will be able to figure out what the code exactly when they look at it.

Also, add more inline comments, if I hid all the code the inline comments should the full story of what you're doing to achieve the goal. More text is not bad.

**Pull request merged**